src/components/TodoItem.jsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from 'react';
+import styles from '../styles/TodoItem.module.css';
+
+const TodoItem = ({
+  todo, handleChange, handleDelete, setUpdate,
+}) => {
+  const [editing, setEditing] = useState(false);
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const viewMode = {};
+  const editMode = {};
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  const handleUpdateDone = (event) => {
+    if (event.key === 'Enter') {
+      setEditing(false);
+    }
+  };
+
+  return (
+    <>
+      <li className={styles.item}>
+        <div className={styles.content} style={viewMode}>
+          <input
+            type="checkbox"
+            checked={todo.completed}
+            onChange={() => handleChange(todo.id)}
+          />
+          <button type="submit" onClick={handleEditing}>Edit</button>
```

Comment:
You may add SVG icon for edit instead of `Edit` text.
Your Design:
![image](https://github.com/manzitresor/react_todo_app/assets/24216198/f139f7df-5b64-475f-bc1c-19f0643a5b2f)
Tutorial Design:
![image](https://github.com/manzitresor/react_todo_app/assets/24216198/612cc1b9-10f8-4a1c-969a-39c079cc2456)